lib/spack/spack/test/conftest.py

```diff
@@ -1980,17 +1980,24 @@ def mock_modules_root(tmp_path, monkeypatch):
     monkeypatch.setattr(spack.modules.common, "root_path", fn)
 
 
+_repo_name_id = 0
+
+
 def create_test_repo(tmpdir, pkg_name_content_tuples):
+    global _repo_name_id
+
     repo_path = str(tmpdir)
     repo_yaml = tmpdir.join("repo.yaml")
     with open(str(repo_yaml), "w") as f:
         f.write(
-            """\
+            f"""\
 repo:
-  namespace: testcfgrequirements
+  namespace: testrepo{str(_repo_name_id)}
 """
         )
 
+    _repo_name_id += 1
```

Comment:
I realized that the test repos I was creating for different unit tests all had the same name, which led to issues with Python imports if a package had the same name in two of those repos (originally I thought it was an issue about clearing provider caches etc., which might still be a good idea, but this was a higher-level issue).